cpp/src/slice2js/Gen.cpp

```diff
@@ -510,47 +489,43 @@ Slice::JsVisitor::writeConstantValue(
     return os.str();
 }
 
-StringList
-Slice::JsVisitor::splitComment(const ContainedPtr& p)
+void
+Slice::JsVisitor::writeDocCommentFor(const ContainedPtr& p, bool includeDeprecated)
 {
-    StringList result;
+    StringList lines;
 
-    string comment = p->comment();
-    string::size_type pos = 0;
-    string::size_type nextPos;
-    string line;
-    while ((nextPos = comment.find_first_of('\n', pos)) != string::npos)
     {
-        line = string(comment, pos, nextPos - pos);
+        string comment = p->comment();
+        string::size_type pos = 0;
+        string::size_type nextPos;
+        string line;
+        while ((nextPos = comment.find_first_of('\n', pos)) != string::npos)
+        {
+            line = string(comment, pos, nextPos - pos);
+            pos = line.find_first_not_of(" \t");
+            if (pos != string::npos)
+            {
+                line = line.substr(pos);
+            }
+            lines.push_back(line);
+            pos = nextPos + 1;
+        }
+
+        line = string(comment, pos);
         pos = line.find_first_not_of(" \t");
         if (pos != string::npos)
         {
             line = line.substr(pos);
         }
-        result.push_back(line);
-        pos = nextPos + 1;
-    }
 
-    line = string(comment, pos);
-    pos = line.find_first_not_of(" \t");
-    if (pos != string::npos)
-    {
-        line = line.substr(pos);
-    }
-
-    if (line.find_first_not_of(" \t\n\r") != string::npos)
-    {
-        result.push_back(line);
+        if (line.find_first_not_of(" \t\n\r") != string::npos)
+        {
+            lines.push_back(line);
+        }
     }
 
-    return result;
-}
-
-void
-Slice::JsVisitor::writeDocCommentFor(const ContainedPtr& p, bool includeDeprecated)
-{
-    StringList lines = splitComment(p);
-    CommentPtr dc = p->parseComment(false);
+    // TODO this whole function seems bogus. Why do we manually split the lines and also call `parseComment`?
```

Comment:
We should switch to use the lines provided by the parsed comment and remove the custom split lines logic.

Reply:
I agree, but there's even more to cleanup/remove around this code.
So, I'm planning on doing it in a separate PR, right after this one gets merged.

There's an endless rabbit-hole of cleanup in the parser, and I need to draw the line somewhere or I'd never get around to opening my PRs : vP